sql/core/src/main/scala/org/apache/spark/sql/execution/SparkPlanInfo.scala

```diff
@@ -33,7 +34,8 @@ class SparkPlanInfo(
     val simpleString: String,
     val children: Seq[SparkPlanInfo],
     val metadata: Map[String, String],
-    val metrics: Seq[SQLMetricInfo]) {
+    val metrics: Seq[SQLMetricInfo],
+    val relation: Seq[SparkPlanInfo] = Seq()) {
```

Comment:
I'm a little worried about adding a new parameter to `SparkPlanInfo`. What's the semantic of it for general cases?

Reply:
I tried to replicate the same structure that in InMemoryTableScan where the relation is a different element and is not a children.
This attribute has a default value so it not impact in any case where is not used. Only in this case is used. Do you think there is a better way to solve it without the new attribute? Thanks!

Reply:
Can we treat `InMemoryTableScan.relation` as its child in the UI?